var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -38,15 +57,26 @@ class Zfp(CMakePackage):
             description='Use a faster but more collision prone hash function')
     variant('profile', default=False,
             description='Count cache misses')
-    variant('shared', default=True,
-            description='Build shared versions of the library')
 
+    # Dependencies
     depends_on('cmake@3.4.0:', type='build')
+    depends_on('cuda@7:',      type=('build', 'test', 'run'), when='+cuda')
+    depends_on('py-numpy',     type=('build', 'test', 'run'), when='+python')
+    depends_on('py-cython',    type='build',                  when='+python')
 
     def cmake_args(self):
         spec = self.spec
 
+        # Common CMake options for zfp 0.5.1 and later
         args = [
+            '-DBUILD_UTILITIES:BOOL={0}'.format(
+                'ON' if '+utilities' in spec else 'OFF'),
```

Comment:
All of these CMake defines can be shortened using convenience methods. For instance the one above can be written as:
```suggestion
           self.define_from_variant('BUILD_UTILITIES', 'utilities'),
```

Reply:
I like it!

Reply:
One note: I just realized that our CMake scripts currently don't install the utilities, and for spack it doesn't make sense to build the examples.  We'll fix this in the next zfp release, so maybe we can leave this line in but commented out?

Reply:
Quick question on how to use this with testing (please forgive my limited knowledge of Spack, CMakePackage, and Python).  Currently, the package lists this: `'-DBUILD_TESTING:BOOL={0}'.format('ON' if self.run_tests else 'OFF')`.  However, I also added a variant `tests` with the intent that this needs to be enabled to build the tests.  In other words, I would want to rewrite this as `self.define_from_variant('BUILD_TESTING', 'tests'), but I understand that `self.run_tests` is set on the command-line via the `--run-tests` Spack option.  Would you advise simply doing away with the tests variant and relying on the Spack option to enable testing?  I'm guessing something like `self.define('BUILD_TESTING', self.run_tests)` would do the trick?

Similarly, I would want to replace `'-DZFP_BIT_STREAM_WORD_SIZE:STRING={0}'.format(spec.variants['bsws'].value)` with `self.define('ZFP_BIT_STREAM_WORD_SIZE', spec.variants['bows'].value)`.  Any concern here?

Reply:
> I'm guessing something like self.define('BUILD_TESTING', self.run_tests) would do the trick

That's correct. The preference would be to turn all the dependencies needed for testing into "test" type dependencies, and rely on `self.run_test`. 

> Similarly, I would want to replace '-DZFP_BIT_STREAM_WORD_SIZE:STRING={0}'.format(spec.variants['bsws'].value) [ ... ] Any concern here?

I'm absolutely fine with that change!